twitter.py

```diff
@@ -160,6 +161,28 @@ def tweet_wrong_fee_block(block_data):
     # Increment global index at the end of the function
     last_twit_index = (last_twit_index + 1) % len(sad_emojis) # Assuming all lists are the same length
 
+def fetch_donations_data(url):
+    try:
+        response = requests.get(url, timeout=10)
+        response.raise_for_status()
+        return response.json()
+    except requests.RequestException as e:
+        logging.error(f"Network error occurred when calling the Donations API: {e}")
+        return None
+
+def tweet_new_donation(donation_data):
+    amount_eth = w3.from_wei(int(donation_data['amount_wei']), 'ether')
+    donor_address = donation_data['donor_address']
+    tweet = f"🎉 New Donation! 🎉\n\n" \
```

Comment:
Only the donation amount and donor address will change. If two or more donations happen in a row, maybe a tweet is flagged as duplicate content and denied by twitter API. We should cycle between 3 or 4 possible copys for donations, like it is done for proposed blocks

Also, we could add extra happy emojis if donation > 1 ETH